java/spring/security/audit/spring-sqli.yaml

```diff
@@ -1,142 +1,36 @@
-                    ...
-                  }
-          - pattern-not: |
-              $JT.batchUpdate(...,"..." + "...",...);
-      - patterns:
-          - pattern-either:
-              - patterns:
-                  - pattern-either:
-                      - pattern-inside: |
-                          String $SQL = $X + $Y;
-                          ...
-                      - pattern-inside: |
-                          String $SQL = String.format(...);
-                          ...
-                      - pattern-inside: |
-                          $VAL $FUNC(...,String $SQL,...) {
-                            ...
-                          }
-                  - pattern-not-inside: |
-                      String $SQL = "..." + "...";
-                      ...
-                  - pattern: new PreparedStatementCreatorFactory($SQL,...)
-              - pattern: |
-                  new PreparedStatementCreatorFactory(String.format(...),...)
-              - pattern: |
-                  new PreparedStatementCreatorFactory($X + $Y,...)
-          - pattern-not: |
-              new PreparedStatementCreatorFactory("..." + "...",...)
-      - patterns:
-          - pattern-either:
-              - patterns:
-                  - pattern-either:
-                      - pattern-inside: |
-                          String $SQL = $X + $Y;
-                          ...
-                      - pattern-inside: |
-                          String $SQL = String.format(...);
-                          ...
-                      - pattern-inside: |
-                          $VAL $FUNC(...,String $SQL,...) {
-                            ...
-                          }
-                  - pattern-not-inside: |
-                      String $SQL = "..." + "...";
-                      ...
-                  - pattern: $BATCH.executeBatchUpdate($SQL,...)
-              - pattern: |
-                  $BATCH.executeBatchUpdate(String.format(...),...)
-              - pattern: |
-                  $BATCH.executeBatchUpdate($X + $Y,...)
-          - pattern-not: |
-              $BATCH.executeBatchUpdate("..." + "...",...)
-          - metavariable-regex:
-              metavariable: $BATCH
-              regex: ^(BatchUpdateUtils|NamedParameterBatchUpdateUtils)$
+    mode: taint
+    pattern-sources:
+    - patterns:
+        - pattern: $ARG
+        - pattern-inside: |
+            public $T $M (..., String $ARG,...){...}
+    pattern-sinks:
+    - patterns:
+        - pattern-either:
+            - patterns:
+                - pattern: $A
+                - pattern-inside: |
+                    new PreparedStatementCreatorFactory($A,...);
+            - patterns:
+                - pattern: $A
+                - pattern-inside: |
+                    (JdbcTemplate $T).$M($A,...)
+            - patterns:
+                - pattern: $A
+                - pattern-inside: |
+                    (JdbcTemplate $T).batchUpdate(...)
```

Comment:
This is due to how `batchUpdate` specifically handles the `(String ... sql)` case: https://github.com/spring-projects/spring-framework/blob/971f665eb1e8cd9e8a0c788a8f5612f815894082/spring-jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java#L579-L583